Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitInspector.cs

```diff
@@ -118,11 +143,267 @@ public override void OnInspectorGUI()
+                        horizontalGroupOpen = true;
+                    }
+
+
+                    bool selected = (config.SelectedSubjects & tag) != 0;
+                    GUI.color = selected ? MixedRealityInspectorUtility.SuccessColor : MixedRealityInspectorUtility.DisabledColor;
+                    if (GUILayout.Button(tag.ToString(), EditorStyles.miniButton))
+                    {
+                        if (selected)
+                        {
+                            config.SelectedSubjects &= ~tag;
+                        }
+                        else
+                        {
+                            config.SelectedSubjects |= tag;
+                        }
+                    }
+                    GUI.color = Color.white;
+
+                    numButtons++;
+                    if (numButtons >= maxSubjectButtonsPerLine)
+                    {
+                        EditorGUILayout.EndHorizontal();
+                        horizontalGroupOpen = false;
+                    }
+                }
+
+                if (horizontalGroupOpen)
+                {
+                    EditorGUILayout.EndHorizontal();
+                }
+            }
+
+            EditorGUILayout.Space();
+            EditorGUILayout.Space();
+
+            EditorGUILayout.HelpBox("Enter keywords below to search for fields.\nIf any subjects are selected, those will be used to narrow your search. Select (None) to search all fields.", MessageType.Info);
+            config.SearchFieldString = EditorGUILayout.TextField(config.SearchFieldString);
+
+            EditorGUILayout.Space();
+            EditorGUILayout.Space();
+
+            #endregion
+
+            #region options
+
+            bool optionsFoldout = SessionState.GetBool(searchDisplayOptionsFoldoutKey, false);
+            optionsFoldout = EditorGUILayout.Foldout(optionsFoldout, "Options", true);
+            if (optionsFoldout)
+            {
+                config.RequireAllKeywords = EditorGUILayout.Toggle("Require All Keywords", config.RequireAllKeywords);
+                config.RequireAllSubjects = EditorGUILayout.Toggle("Require All Tags", config.RequireAllSubjects);
+                config.SearchTooltips = EditorGUILayout.Toggle("Search Tooltips", config.SearchTooltips);
+                config.SearchFieldObjectNames = EditorGUILayout.Toggle("Search GameObject Names", config.SearchFieldObjectNames);
+                config.SearchChildProperties = EditorGUILayout.Toggle("Search Child Properties", config.SearchChildProperties);
+            }
+
+            EditorGUILayout.Space();
+            EditorGUILayout.Space();
+
+            #endregion
+
+            SessionState.SetString(searchDisplaySearchFieldKey, config.SearchFieldString);
+            SessionState.SetString(searchDisplaySelectedTagsKey, MixedRealitySearchUtility.SubjectFlagsToString(config.SelectedSubjects));
+            SessionState.SetBool(searchDisplayRequireAllTagsKey, config.RequireAllSubjects);
```

Comment:
And this with SearchConfigInspector utility above...or again move better name :P